graph-builder/src/registry.rs

```diff
@@ -293,3 +293,144 @@ fn assemble_metadata(blob: &[u8], metadata_filename: &str) -> Result<Metadata, E
     }
     .map_err(Into::into)
 }
+
+#[cfg(test)]
+mod tests {
+
+    #[cfg(feature = "test-net")]
+    mod tests_net {
```

Comment:
These are integration tests and should be in the `tests` directory instead.

Reply:
@crawford PTAL at the last commit which implements your suggestion. It's quite invasive but that's fine with me.